server/blob/blob-s3/src/test/java/org/apache/james/blob/objectstorage/aws/S3WithMinIOGenerationAwareBlobIdTest.java

```diff
@@ -0,0 +1,189 @@
+            .defaultBucketName()
+            .deduplication();
+    }
+
+    @ParameterizedTest
+    @MethodSource("storagePolicies")
+    void saveShouldReturnBlobIdOfString(BlobStore.StoragePolicy storagePolicy) {
+        // Given: A BlobStore and its default bucket
+        BlobStore store = testee();
+        BucketName defaultBucketName = store.getDefaultBucketName();
+
+        // When: Saving a blob
+        BlobId blobId = Mono.from(store.save(defaultBucketName, "toto", storagePolicy)).block();
+        String blobIdString = blobId.asString();
+
+        // Then: BlobId string and parsed BlobId should match expectations
+        assertThat(blobIdString).isEqualTo("1/628/M/f/e/m/XjFVhqwZi9eYtmKc5JA9CJlHbVdBqfMuLlIbamY=");
+        assertThat(blobId).isEqualTo(blobIdFactory().parse(blobIdString));
+    }
+
+    @Test
+    @Override
+    @Disabled("S3minio return `Connection: close` in header response, https://github.com/apache/james-project/pull/1981#issuecomment-2380396460")
+    public void deleteBucketConcurrentlyShouldNotFail() {
+    }
+
+    @Nested
+    class Compatible {
+
+        private BlobStore withGenerationAwareBlobId;
+        private BlobStore withMinIOGenerationAwareBlobId;
+        private BucketName defaultBucketName;
+
+        @BeforeEach
+        void setup() {
+            BlobId.Factory plainBlobIdFactory = new PlainBlobId.Factory();
+            withGenerationAwareBlobId = createBlobStore(new GenerationAwareBlobId.Factory(clock, plainBlobIdFactory, GenerationAwareBlobId.Configuration.DEFAULT));
+            withMinIOGenerationAwareBlobId = createBlobStore(new MinIOGenerationAwareBlobId.Factory(clock, GenerationAwareBlobId.Configuration.DEFAULT, plainBlobIdFactory));
+            defaultBucketName = withGenerationAwareBlobId.getDefaultBucketName();
+        }
+
+
+        @Test
+        void readWithMinIOGenerationAwareShouldSuccessWhenBlobWasStoredByGenerationAware() {
+            String originalData = "toto" + UUID.randomUUID();
+            // Given a blob stored with GenerationAwareBlobId
+            BlobId blobId = Mono.from(withGenerationAwareBlobId.save(defaultBucketName, originalData, LOW_COST)).block();
+
+            assertThat(blobId).isInstanceOf(GenerationAwareBlobId.class);
+
+            // When reading it with MinIOGenerationAwareBlobId
+            byte[] readAsByte = Mono.from(withMinIOGenerationAwareBlobId.readBytes(defaultBucketName, blobId)).block();
+
+            // Then the data should be the same
+            assertThat(new String(readAsByte, StandardCharsets.UTF_8)).isEqualTo(originalData);
+        }
+
+        @Test
+        void listBlobsShouldReturnCorrectBlobIdWhenBlobWasStoredByGenerationAware() {
+            String originalData = "toto" + UUID.randomUUID();
+            // Given a blob stored with GenerationAwareBlobId
+            BlobId blobId = Mono.from(withGenerationAwareBlobId.save(defaultBucketName, originalData, LOW_COST)).block();
```

Comment:
Let's do the reverse also: save with `withMinIOGenerationAwareBlobId` and read with `withGenerationAwareBlobId`

This will ensure we can revert if we need...

Reply:
added